### PR TITLE
RavenDB-8862 Add Delete ongoing task for Subscriptions with user authentication

### PR DESCRIPTION
**Pull request by IdanHaim**

### CODE REVIEW

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs [hunk 1]
```diff
@@ -574,6 +574,21 @@ private void WriteResult(JsonOperationContext context, DynamicJsonValue dynamicJ
...
+
+        [RavenAction("/databases/*/subscription-tasks", "DELETE", AuthorizationStatus.ValidUser)]
+        public async Task DeleteSubscriptionTask()
+        {
+            // Note: Only Subscription task needs User authentication, All other tasks need Admin authentication
+            var typeStr = GetQueryStringValueAndAssertIfSingleAndNotEmpty("type");
+            if (Enum.TryParse<OngoingTaskType>(typeStr, true, out var type) == false)
+                throw new ArgumentException($"Unknown task type: {type}", nameof(type));
+
+            if (type != OngoingTaskType.Subscription)
```
Comment (ml054) [inline src/Raven.Server/Web/System/OngoingTasksHandler.cs:586]: what if you pass different task type? Then you can remove subscription w/o access? 
I think we should check here if task with given type is actually subscription.

Comment (ayende) [reply src/Raven.Server/Web/System/OngoingTasksHandler.cs:586]: If you pass anything else it will throw, which is what we want, no?

Comment (ml054) [reply src/Raven.Server/Web/System/OngoingTasksHandler.cs:586]: what if some ETL has task id:5 and I say:
DELETE /databases/*/subscription-tasks?type=Subscription&taskId=5

Will it delete etl task?

File: src/Raven.Studio/typescript/commands/database/tasks/deleteOngoingTaskCommand.ts [hunk 2]
```diff
@@ -10,12 +10,16 @@ class deleteOngoingTaskCommand extends commandBase {
...
-        const url = endpoints.databases.ongoingTasks.adminTasks + this.urlEncodeArgs(args);
 
-        return this.del<Raven.Client.ServerWide.Operations.ModifyOngoingTaskResult>(url, null, this.db)
+        // Subscription is the only task that needs only *User* authenication for delete task
+        // All others use *Admin* authentication
+        const url = this.taskType === "Subscription" ? endpoints.databases.ongoingTasks.subscriptionTasks :
+                                                       endpoints.databases.ongoingTasks.adminTasks;
+
+        return this.del<Raven.Client.ServerWide.Operations.ModifyOngoingTaskResult>(url + this.urlEncodeArgs(args), null, this.db)
             .done(() => this.reportSuccess(`Successfullly deleted ${this.taskType} task`))
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/database/tasks/deleteOngoingTaskCommand.ts:20]: here we have typo: Successfullly

File: src/Raven.Server/Web/System/OngoingTasksHandler.cs (hunk 1 above)
Comment (ml054) [reply src/Raven.Server/Web/System/OngoingTasksHandler.cs:586]: ok I can see we get task type again in: DeleteOngoingTask method, so it shouldn't be issue. Please ignore my comments then.

**Pull request merged**